src/Middleware/ErrorResponseGenerator.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive for the canonical source repository
+ * @copyright Copyright (c) 2016 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Expressive\Middleware;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Zend\Expressive\Template\TemplateRendererInterface;
+use Zend\Stratigility\Utils;
+
+class ErrorResponseGenerator
+{
+    const TEMPLATE_DEFAULT = 'error::error';
+
+    /**
+     * @var bool
+     */
+    private $debug = false;
+
+    /**
+     * @var TemplateRendererInterface
+     */
+    private $renderer;
+
+    /**
+     * @var string
+     */
+    private $stackTraceTemplate = <<<'EOT'
+%s raised in file %s line %d:
+Message: %s
+Stack Trace:
+%s
+
+EOT;
+
+    /**
+     * @var string
+     */
+    private $template;
+
+    /**
+     * @param bool $isDevelopmentMode
+     * @param null|TemplateRendererInterface $renderer
+     * @param string $template
+     */
+    public function __construct(
+        $isDevelopmentMode = false,
+        TemplateRendererInterface $renderer = null,
+        $template = self::TEMPLATE_DEFAULT
+    ) {
+        $this->debug     = (bool) $isDevelopmentMode;
+        $this->renderer  = $renderer;
+        $this->template  = $template;
+    }
+
+    /**
+     * @param \Throwable|\Exception $e
+     * @param ServerRequestInterface $request
+     * @param ResponseInterface $response
+     * @return ResponseInterface
+     */
+    public function __invoke($e, ServerRequestInterface $request, ResponseInterface $response)
```

Comment:
Except that `\Throwable` does not inherit from `\Exception` in PHP 7, which would make things like `TypeError` `$e` values create _new_ `TypeError`s.

This is a necessary evil until we can move to PHP 7+ only.